src/pages/userManagement/components/EditCollaborationDialog.jsx

```diff
@@ -0,0 +1,224 @@
+import React, { useEffect, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+
+import Alert from '../../../components/Alert';
+import Button from '../../../components/Button';
+import InputRow from '../../../components/fields/edit/InputRow';
+import SelectionEditor from '../../../components/fields/edit/SelectionEditor';
+import StandardDialog from '../../../components/StandardDialog';
+import Text from '../../../components/Text';
+import usePatchCollaboration from '../../../models/collaboration/usePatchCollaboration';
+import {
+  getSummaryState,
+  isEditApproved,
+  isViewApproved,
+  isViewRevoked,
+  states as collaborationStates,
+  summaryStates,
+} from '../../../models/collaboration/model';
+
+const coreStateChoices = [
+  {
+    value: 'view',
+    labelId: 'COLLABORATION_STATE_VIEW',
+  },
+  {
+    value: 'edit',
+    labelId: 'COLLABORATION_STATE_EDIT',
+  },
+];
+
+const fullStateChoices = [
+  ...coreStateChoices,
+  {
+    value: 'revoked',
+    labelId: 'COLLABORATION_STATE_REVOKED',
+  },
+];
+
+function FormattedUserName({ name }) {
+  return name || <FormattedMessage id="UNNAMED_USER" />;
+}
+
+function getPatchCollaborationOperations(
+  newSummaryState,
+  collaboration,
+) {
+  switch (newSummaryState) {
+    case summaryStates.revoked:
```

Comment:
switch statements can be really prone to bugginess. this one looks okay but in general i prefer if statements (even if it doesn't look quite as good)

Reply:
Changed to `if` statements in https://github.com/WildMeOrg/codex-frontend/pull/451/commits/466d459bf84ef3ece5a0152ab78512b40070529c.